### PR TITLE
Handle empty content message responses

### LINKED ISSUE #2486: HTTP HEAD does not return a response to the client
Step 1 - Set up a back end service that accepts http head as the http method for the resource
Step 2 - Write a similar code as given below

```
import ballerina.net.http;

@http:BasePath {value:"/http"}
service contentBasedRouting {
    string connection = "http://localhost:5301";

    @http:POST {}
    @http:GET {}
    @http:Path {value:"/callHead"}
    resource httpHeadResource (message m) {
        string resourcePath = "/abc";
        string headerValue;
        message response;
        message request = m;
        http:ClientConnector httpCheck = create http:ClientConnector(connection);
        response = http:ClientConnector.head(httpCheck, resourcePath, request);
        reply response;
    }

}

```

Step 3. Invoke the above service

**Observations**
1. When the communication is analyzed using tcp mon, below was seen that indicates, outgoing and incoming from and to ballerina

**Incoming message from client to ballerina**

```
POST /http/callHead HTTP/1.1
Host: localhost:5302
Connection: keep-alive
Content-Length: 122
Postman-Token: 81fe162a-0a87-173b-6cc4-9334f5faa477
Cache-Control: no-cache
Origin: chrome-extension://fhbjgbiflinjbdggehcddcbncdddomop
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36
Content-Type: application/json
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8

{
   "Order":{
       "orderId":"ee1a9ec2-c8a5-4afe-8585-74df591f9990",
       "additions":"Chocolate Chip Cookies"
   }
}
```

**Outgoing from ballerina**
```
HEAD /abc HTTP/1.1
Connection: keep-alive
Content-Length: 122
Postman-Token: 0c5da918-1b64-c2bb-f384-b8a420967f5b
Cache-Control: no-cache
Origin: chrome-extension://fhbjgbiflinjbdggehcddcbncdddomop
Content-Type: application/json
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8
HOST: localhost:5301
User-Agent: ballerina/0.88-SNAPSHOT

{
   "Order":{
       "orderId":"ee1a9ec2-c8a5-4afe-8585-74df591f9990",
       "additions":"Chocolate Chip Cookies"
   }
}
```

**Incoming to Ballerina**
```
HTTP/1.1 200 OK
Content-Type: application/json
Content-Encoding: gzip
Content-Length: 145
Server: Jetty(6.1.26)

```
**Outgoing from ballerina to client** 
```
None
```

### PR DESCRIPTION
**Pull request by chamil321**
This PR will resolve #2486

### CODE REVIEW

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/actions/http/AbstractHTTPAction.java [hunk 1]
```diff
@@ -198,6 +198,17 @@ public void validate(BalConnectorCallback callback) {
         handleTransportException(callback.getValueRef());
     }
 
+    // Handle operations for empty content messages initiated from the backend service
```
Comment (hasithaa) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/actions/http/AbstractHTTPAction.java]: Add propper java doc comment.

**Comment by chamil321** [comment]
@hasithaa I added integration test cases to tools-distribution

**Comment by hasithaa** [comment]
@chamil321 https://github.com/ballerinalang/tools-distribution/pull/120 has conflicts. Can you resolve it?

**Comment by chamil321** [comment]
This PR is no longer required as the change is done at carbon transport under https://github.com/wso2/carbon-transports/pull/345

**Comment by hasithaa** [comment]
Closing as per comment.

**Pull request closed without merging**